src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -158,77 +205,55 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Extensions
 
-(defn get-extensions
+(schema/defn ^:always-validate get-extensions :- ExtensionList
   "Given an object containing X509 extensions, retrieve a list of maps of all
   extensions. Each map in the list contains the following keys:
 
   `oid`      : The OID of the extension
   `value`    : The value of the extensions
   `critical` : True if this is a critical extensions, false if it is not."
-  [ext-container]
-  {:pre [(or (certificate? ext-container)
-             (certificate-request? ext-container)
-             (certificate-revocation-list? ext-container))]
-   :post [(extension-list? %)]}
+  [ext-container :- (schema/cond-pre X509Certificate PKCS10CertificationRequest X509CRL)]
   (-> (or (ExtensionsUtils/getExtensionList (javaize ext-container))
           [])
       clojureize))
 
-(defn get-extension
+(schema/defn ^:always-validate get-extension :- Extension
   "Given a X509 certificate object, CRL, CSR, or a list of extensions
   returned by `get-extensions`, return a map describing the value and
   criticality of the extension described by its OID."
-  [ext-container oid]
-  {:pre [(or (certificate? ext-container)
-             (certificate-request? ext-container)
-             (certificate-revocation-list? ext-container)
-             (instance? List ext-container))
-         (string? oid)]
-   :post [(extension? %)]}
+  [ext-container :- ExtensionContainer
+   oid :- schema/Str]
   (-> (ExtensionsUtils/getExtension (javaize ext-container) oid)
       clojureize))
 
-(defn get-extension-value
+(schema/defn ^:always-validate get-extension-value
```

Comment:
I suppose you could use schema on the return value here for visual purposes even though it could be anything.